tests/migration_test.go

```diff
@@ -931,24 +932,59 @@ var _ = Describe("[rfe_id:393][crit:high][vendor:cnv-qe@redhat.com][level:system
 		})
 		Context("with an Fedora shared NFS PVC, cloud init and service account", func() {
 			var pvName string
+			var nfsPodName string
 			var vmi *v1.VirtualMachineInstance
-			BeforeEach(func() {
+
+			initNFS := func(ipFamily k8sv1.IPFamily) {
 				tests.SkipNFSTestIfRunnigOnKindInfra()
 				pvName = "test-nfs" + rand.String(48)
 				// Prepare a NFS backed PV
 				By("Starting an NFS POD")
 				os := string(cd.ContainerDiskFedora)
-				nfsIP := tests.CreateNFSTargetPOD(os)
+
 				// create a new PV and PVC (PVs can't be reused)
 				By("create a new NFS PV and PVC")
+				nfsPod := tests.CreateNFSTargetPOD(os)
+				nfsPodName = nfsPod.Name
+				nfsIP := libnet.GetPodIpByFamily(nfsPod, ipFamily)
+				ExpectWithOffset(1, nfsIP).NotTo(BeEmpty())
 				tests.CreateNFSPvAndPvc(pvName, "5Gi", nfsIP, os)
+			}
+
+			BeforeEach(func() {
+				pvName = ""
+				nfsPodName = ""
+				vmi = nil
 			})
 
 			AfterEach(func() {
-				// PVs can't be reused
-				tests.DeletePvAndPvc(pvName)
+				if vmi != nil {
```

Comment:
I'm wondering if we should perform the deletion on `AfterEach` conditionally (as you're doing), versus do it in the tests where it must happen. 

I would like to coordinate with you in this because #4224 is raising the same question (and I chose the other route :) ).